### ISSUE TITLE
support api to fetch emails,or push email to other api

### ISSUE THREAD

**Issue by zhaojunlike** [Mailu/Mailu#1239]
such as mail delivery to the designated API, or fetch mails

**Comment by kaiyou** [comment]
We have a project to support a management api for our config, accounts, etc.

But mail fetching and sending needs to be handled in a standard manner, which is why we use imap and smtp.

We have an open issue about implementing jmap also. But we will not implement specific web api for sending or fetching emails.

**Comment by zhaojunlike** [comment]
thx you reply, In many situations, the HTTP protocol can satisfy many scenarios，Now I need loop databases or forwarding, and i can't send email use http, so ....

**Comment by kaiyou** [comment]
I am sorry, I did not understand your use case. Could you rephrase it somehow?

**Comment by zhaojunlike** [comment]
em...., Now I use a forward mail server, It does not handle any mail, just  forwarding to my gmail, and now,But it is deployed very difficult , i want use mailu to set a httpAPI, same as github tragger, if it recive a email, it push data to my http api

**Comment by zhaojunlike** [comment]
My English expression is not very good, I'm terribly sorry。 and  i can't find any mail server support http push trigger , I tried to modify the code but it is difficult for me

**Comment by zhaojunlike** [comment]
I want to do a magic mailbox, Don't let others know my real email address. and i can use *@demo.com any email address

**Comment by zhaojunlike** [comment]
bro, thanks ,thx thx, thx your reply。

**Comment by kaiyou** [comment]
Unfortunately, what you are attempting is not in the scope of Mailu, which is about providing a mail server, not an smtp-http gateway.

If you want this implemented I would suggest you either find an smtp implementation that is simple enough to just hack with it and add the http forwarding. Or you could code a bot that just polls IMAP accounts and then sends emails to http, probably fetchmail can be of help, or you can write it from scratch using Python for instance.

If you really want to patch it through using Mailu, I believe you could rely on sieve scripts. Those cannot be edited from the web ui yet, but you can access the Jinja template in the admin container. There, you can use external scripts to handle your email, including calls to curl for instance, that would forward your email through http.

**Comment by kaiyou** [comment]
Also, if you succeed, this kind of use of sieve scripts could be documented for others to use.

**Comment by zhaojunlike** [comment]
ok thanks, em....  i close the issue
